agents/cryomech-cpa1114/cryomech_cpa1114_agent.py

```diff
@@ -35,7 +35,73 @@ def signal_handler(signum, frame):
         yield
     finally:
         signal.alarm(0)
+
+class PTC:
+    def __init__(self, ip_address, port=502, timeout=10):
+        self.ip_address = ip_address
+        self.port = port  
+        
+        self.comm = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.comm.connect((self.ip_address, self.port))
+        self.comm.settimeout(timeout)
+                 
+    def get_data(self):
+        """
+        Gets the raw data from the ptc and returns it in a usable format. 
+        """
+        self.comm.sendall(self.buildRegistersQuery()) 
+        data = self.comm.recv(1024)
+        brd = self.breakdownReplyData(data)
+            
+        return brd    
+    
+    def buildRegistersQuery(self):
+        query = bytes([0x09, 0x99,  # Message ID
+                       0x00, 0x00,  # Unused
+                       0x00, 0x06,  # Message size in bytes
+                       0x01,        # Slave Address
+                       0x04,        # Function Code  3= Read HOLDING registers, 4 read INPUT registers
+                       0x00,0x01,   # The starting Register Number
+                       0x00,0x35])  # How many to read
+        return query
+
+    def breakdownReplyData(self, rawdata):
+        """
+        Take in raw ptc data, and return a dictionary. The dictionary keys are the data labels, 
+        the dictionary values are the data in floats or ints. 
+        """
+        
+        #Associations between keys and their location in rawData
+        keyloc = {"Operating State": [9,10], "Pump State": [11, 12], "Warnings": [15, 16, 13, 14], "Alarms":[19, 20, 17, 18],
+                  "Coolant In": [22, 21, 24, 23], "Coolant Out": [26, 25, 28, 27], "Oil": [30, 29, 32, 31], 
+                  "Helium": [34, 33, 36, 35], "Low Pressure": [38, 37, 40, 39], "Low Pressure Average":[42, 41, 44, 43],
+                  "High Pressure": [46,45,48,47], "High Pressure Average": [50, 49, 52, 51], "Delta Pressure": [54, 53, 56, 55],
+                  "Motor Current": [58, 57, 60, 59]}
         
+        #Iterate through all keys and return the data in a usable format
+        data = {}
+        for key in keyloc.keys():
+            locs = keyloc[key] 
+            wkrBytes = bytes([rawdata[loc] for loc in locs])
+            
+            #three different data formats to unpack
+            if key in list(keyloc.keys())[0:2]:
```

Comment:
dict keys are not ordered, so this is an unreliable way of selecting keys. Instead, I would do something like:
```
if key in ["Coolant In", "Coolant Out", "Oil"]:
   do something
```

Reply:
Made this change